libraries/tock-cells/src/volatile_cell.rs

```diff
@@ -1,29 +1,41 @@
 //! Implementation of a type for accessing MCU registers.
 
+use core::cell::Cell;
 use core::ptr;
 
 /// `VolatileCell` provides a wrapper around unsafe volatile pointer reads
 /// and writes. This is particularly useful for accessing microcontroller
 /// registers.
-// Source: https://github.com/hackndev/zinc/tree/master/volatile_cell
-#[derive(Copy, Clone, Default)]
-#[repr(C)]
-pub struct VolatileCell<T> {
-    value: T,
+// Originally modified from: https://github.com/hackndev/zinc/tree/master/volatile_cell
+#[derive(Default)]
+#[repr(transparent)]
+pub struct VolatileCell<T: ?Sized> {
+    value: Cell<T>,
```

Comment:
nit: We don't use `Cell`'s ops here, is there a reason it was chosen over `UnsafeCell`?

Reply:
Because I thought it saved us some custom implementations and unsafes but it doesn't. My bad :/